src/integrationTests/basic.test.ts

```diff
@@ -529,9 +539,17 @@ testCookieAndLocalStorage(() => {
     });
 
     describe("when token refresh returns optout", () => {
-      beforeEach(() => {
-        xhrMock.responseText = btoa(JSON.stringify({ status: "optout" }));
-        xhrMock.onreadystatechange(new Event(""));
+      beforeEach(async () => {
+        try {
+          advertisingTokenRefreshedPromise = uid2.getAdvertisingTokenAsync();
+          await xhrMock.sendEncodedRefreshApiResponse(
+            "optout",
+            originalIdentity.refresh_response_key
+          );
+          await advertisingTokenRefreshedPromise;
```

Comment:
I've seen this pattern in a lot of tests - why do we need to do this now, when we didn't before?